iceoryx_utils/include/iceoryx_utils/internal/cxx/expected.inl

```diff
@@ -346,6 +346,58 @@ expected<ValueType, ErrorType>::and_then(const cxx::function_ref<void(ValueType&
     return *this;
 }
 
+template <typename ValueType, typename ErrorType>
+template <typename Optional, typename std::enable_if<is_optional<Optional>::value, int>::type>
+inline const expected<ValueType, ErrorType>&
+expected<ValueType, ErrorType>::and_then(const cxx::function_ref<void(typename Optional::type&)>& callable) const
+    noexcept
+{
+    return const_cast<expected*>(this)->and_then(callable);
+}
+
+template <typename ValueType, typename ErrorType>
+template <typename Optional, typename std::enable_if<is_optional<Optional>::value, int>::type>
```

Comment:
There is also enable_if_t (C++14) which can be used to shorten this a bit (not required).

Reply:
This was already added to our cxx/type_trait.hpp and can be removed once we switched to C++14

Reply:
Ah, I forgot about this.